python-checks/src/main/java/org/sonar/python/checks/cdk/DisabledRDSEncryptionCheck.java

```diff
@@ -0,0 +1,91 @@
+  private static final String ARG_ENCRYPTED = "storage_encrypted";
+  private static final String ARG_ENCRYPTION_KEY = "storage_encryption_key";
+  private static final String DB_OMITTING_MESSAGE = "Omitting \""+ARG_ENCRYPTED+"\" and \""+ARG_ENCRYPTION_KEY+"\" disables RDS encryption. Make sure it is safe here.";
+  private static final String CFNDB_OMITTING_MESSAGE = "Omitting \""+ARG_ENCRYPTED+"\" disables RDS encryption. Make sure it is safe here.";
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_rds.DatabaseCluster", this::checkDatabaseArguments);
+    checkFqn("aws_cdk.aws_rds.DatabaseInstance", this::checkDatabaseArguments);
+    checkFqn("aws_cdk.aws_rds.CfnDBCluster", this::checkCfnDatabaseArguments);
+    checkFqn("aws_cdk.aws_rds.CfnDBInstance", (subscriptionContext, callExpression) -> {
+      if (!isEngineAurora(subscriptionContext, callExpression)) {
+        checkCfnDatabaseArguments(subscriptionContext, callExpression);
+      }
+    });
+  }
+
+  protected void checkDatabaseArguments(SubscriptionContext ctx, CallExpression resourceConstructor) {
+    Map<String, ArgumentTrace> map = getArguments(ctx, resourceConstructor,
+      List.of(ARG_ENCRYPTED, ARG_ENCRYPTION_KEY));
+    ArgumentTrace argEncrypted = map.get(ARG_ENCRYPTED);
+    ArgumentTrace argEncryptionKey = map.get(ARG_ENCRYPTION_KEY);
+
+    if (argEncrypted == null && argEncryptionKey == null) {
+      ctx.addIssue(resourceConstructor.callee(), DB_OMITTING_MESSAGE);
+    } else if (argEncrypted == null) {
+      argEncryptionKey.addIssueIf(AbstractCdkResourceCheck::isNone, UNENCRYPTED_MESSAGE);
+    } else if (argEncryptionKey == null) {
+      argEncrypted.addIssueIf(AbstractCdkResourceCheck::isFalse, UNENCRYPTED_MESSAGE);
+    } else {
+      if (argEncryptionKey.hasExpression(AbstractCdkResourceCheck::isNone)
+        && argEncrypted.hasExpression(AbstractCdkResourceCheck::isFalse)) {
+        argEncrypted.addIssue(UNENCRYPTED_MESSAGE);
+      }
+    }
+  }
+
+  protected void checkCfnDatabaseArguments(SubscriptionContext ctx, CallExpression resourceConstructor) {
+    getArgument(ctx, resourceConstructor, ARG_ENCRYPTED).ifPresentOrElse(
+      argumentTrace -> argumentTrace.addIssueIf(AbstractCdkResourceCheck::isFalse, UNENCRYPTED_MESSAGE),
+      () -> ctx.addIssue(resourceConstructor.callee(), CFNDB_OMITTING_MESSAGE)
+    );
+  }
+
+  protected boolean isEngineAurora(SubscriptionContext ctx, CallExpression resourceConstructor) {
+    return getArgument(ctx, resourceConstructor, "engine")
+      .filter(argumentTrace ->
+        argumentTrace.hasExpression(expression ->
+          Optional.of(expression)
+            .filter(expr -> expr.is(Tree.Kind.STRING_LITERAL)).map(StringLiteral.class::cast)
+            .filter(stringLiteral -> stringLiteral.trimmedQuotesValue().toLowerCase(Locale.ROOT).startsWith("aurora"))
+            .isPresent()
+        )
+      ).isPresent();
+  }
+}
```

Comment:
```suggestion
  protected boolean isEngineAurora(SubscriptionContext ctx, CallExpression resourceConstructor) {
    return getArgument(ctx, resourceConstructor, "engine")
      .filter(argumentTrace -> argumentTrace.hasExpression(startsWith("aurora"))).isPresent();
  }

  protected Predicate<Expression> startsWith(String expected) {
    return e -> getStringValue(e).filter(expected::startsWith).isPresent();
  }

  protected Optional<String> getStringValue(Expression expression) {
    try {
      return Optional.of(((StringLiteral) expression).trimmedQuotesValue());
    } catch (ClassCastException e) {
      return Optional.empty();
    }
  }
```
An idea to make it more readable.